### PR TITLE
Add ds1803 digital potentiometer support.

### PR DESCRIPTION
**Pull request by andreaskuster**
Add support for the DS1803 I2C digital potentiometer (i.e. to drive the motor controller).

### CODE REVIEW

**Comment by andreaskuster** [comment]
TODO: fix pipeline (seems to be a platformio and not code-related issue)

**Comment by andreaskuster** [comment]
Pipeline has been fixed with PR #30

**Comment by andreaskuster** [comment]
TODO: Test it with the physical board

**Comment by andreaskuster** [comment]
Physically tested & works :)

**Comment by andreaskuster** [comment]
@KarlSchu I think that we should merge this PR and open a separate issue to address the hardware related problem due to two reasons:
- Rebase and merges are getting quite ugly due to the long commit history
- We basically checked that the code works.

File: lib/DAC/DAC.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    while(1) {
+
+        // read user input
+        fgets_stdio_blocking(input_str, n);
+
+        printf("fgets read: %s\n", input_str);
+
+        sscanf(input_str, "%d", &input);
+
+        if(input >= 0 && input < 255){
```
Comment (KarlSchu) [inline lib/DAC/DAC.cpp:58]: Why not 256?

**Review by KarlSchu — APPROVED** [review]

**Review by KarlSchu — APPROVED** [review]

File: lib/DAC/DAC.cpp (hunk 1 above)
Comment (andreaskuster) [reply lib/DAC/DAC.cpp]: Good catch! 
Yes you are right, that should either be a `<=` or `256`

**Comment by andreaskuster** [comment]
Digital potentiometer still fails. Check with Roli about old setup.

**Pull request merged**